### PR TITLE
Convert Radio to typescript & storybook format

### PR DESCRIPTION
**Pull request by NickDunkman**
Presenting this `Radio` button conversion as a POC for our component doc style.

* There is a nice `mdx` file shown under the "Docs" tab that explains everything in sensible groupings.
* The `mdx` file contains `<Story>` definitions that also magically appear on the sidebar -- you can toy with the props for each in the "Canvas" tab.
* Storybook does not by default include all the native `<input>` props, which are included in the `Radio` component props. Rightly so, as there are hundreds of them, and they'd make the storybook docs very noisy. I decided to re-define the types of the more commonly used props in our own `RadioProps` type, so that they can appear more quickly in vscode (without having to recursively click through several prop definitions, have better doc strings, and so that they appear in the props table at the top of the docs.

Screen Shot 2022-01-19 at 15 06 10

### CODE REVIEW

File: src/components/Radio/Radio.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+<Meta title="Components/Radio" component={Radio} />
+
+# Radio
+
+### Props
+
+Below are some of the commonly used props used with `Radio`, but it takes all
+props that can be used with
+[HTMLInputElement](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.htmlinputelement.html)
```
Comment (NickDunkman) [inline src/components/Radio/Radio.stories.mdx:14]: This seemed to be the most authoritative/permanent place to link to for docs on `HTMLInputElement` -- do you know of anything better?

**Comment by NickDunkman** [comment]
Just added a commit that brings the component description we had in the README file, and pulls from component's TSDoc directly into the docs page: https://github.com/aminohealth/phenotypes/pull/172/commits/23ca39a051b11e82b2bec4998c956df8a65b2b41

Screen Shot 2022-01-19 at 15 35 40

**Comment by NickDunkman** [comment]
Sorry, one more! Got the `children` prop to appear in the props table: https://github.com/aminohealth/phenotypes/pull/172/commits/e121733acee47a57c06e6cb3e2168849ecbcd00c

Screen Shot 2022-01-19 at 15 49 39

File: public_api.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ export { default as Button } from './library/components/button/button.jsx';
 export { default as Checkbox } from './library/components/checkbox/checkbox.jsx';
 export { default as FormGroup } from './library/components/form_group/form_group.jsx';
 export { default as ProgressBar } from './library/components/progress_bar/progress_bar.jsx';
-export { default as Radio } from './library/components/radio/radio.jsx';
+export { default as Radio } from './src/components/Radio';
```
Comment (NickDunkman) [inline public_api.js:9]: haven't tested this yet -- have a task to circle back and get all the npm publishing stuff tidied back up at the end of this project

File: src/components/Radio/index.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+export { default } from './Radio';
```
Comment (ahmedmodan) [inline src/components/Radio/index.ts:1]: I think we should avoid default exports altogether. I think we had a discussion about this in slack semi-recently too.  What are your thoughts? re-exporting a default export from another file as a default export from this file feels odd to me. but open to what others think.

File: library/components/radio/_radio.jsx [hunk 4]
```diff
@@ -5,13 +5,6 @@ import Button from '../button/button.jsx';
 function RadioButtonsExamples() {
```
[the remark this answers is not in the document]
Comment (NickDunkman) [reply library/components/radio/_radio.jsx:5]: yea the remaining code here is related to `RadioGroup` -- I'm de-locating `Radio` & `RadioGroup` a bit, so will convert `RadioGroup` later.

File: public_api.js (hunk 2 above)
Comment (NickDunkman) [reply public_api.js]: Ended up testing this when doing the formik docs -- is working 👍

File: src/components/Radio/Radio.stories.mdx (hunk 1 above)
Comment (NickDunkman) [reply src/components/Radio/Radio.stories.mdx]: 👍  Updated: https://github.com/aminohealth/phenotypes/pull/172/commits/1de4b987ba07a48ae0e672502141f1613bb02353

File: src/components/Radio/index.ts (hunk 3 above)
Comment (NickDunkman) [reply src/components/Radio/index.ts]: Yea agreed -- seems safer. Updated here: https://github.com/aminohealth/phenotypes/pull/172/commits/ae8e49a4d2d49a66b59877077619e8953a7e0a6e

Going to set up eslint in a separate PR and will include a rule that errors out on default exports.

**Comment by NickDunkman** [comment]
@ahmedmodan addressed all your comments -- also I added formik docs, as discussed:

ezgif-2-de5c9b40c9

**Review by ahmedmodan — APPROVED** [review]

**Pull request merged**